api/admin/controller/library_settings.py

```diff
@@ -216,29 +221,83 @@ def get_extra_geographic_information(self, value):
 
         return value
 
-    def library_configuration_settings(self, library, validator):
-        for setting in Configuration.LIBRARY_SETTINGS:
-            if setting.get("format") == "geographic":
-                locations = validator.validate_geographic_areas(self.list_setting(setting), self._db)
-                if isinstance(locations, ProblemDetail):
-                    return locations
-                value = locations or self.current_value(setting, library)
-            elif setting.get("type") == "list":
-                value = self.list_setting(setting) or self.current_value(setting, library)
```

Comment:
This is the only place where I may have changed the old behavior during refactoring. I don't think there's actually a behavior change but I want to point it out and get your ghouthts.

I made the change because I thought I saw a bug: I thought this would prevent you from changing a list value from a non-empty list to an empty list. If you tried to do that, `list_setting` would return an empty list, and the empty list -- which the user had explicitly asked for -- would be replaced by the current value.

But then I saw that `list_setting` doesn't return the list, it returns a string containing a JSON-encoded list. That string will never be empty, so the `or self.current_value` here will never run, even if you do pass in an empty list.

So: `self.list_setting(setting)` will always get you something that evaluates to True, so the default will never be used. But, I don't really like the JSON-encoding thing and would like to eventually change it. If we do change it we'll need to make sure we can still handle the case where someone tries to set a non-empty list to an empty list.

Reply:
I don't even remember why we had to JSON-encode it--do you?  Yes, it would be nice to eventually change this (and/or to handle the empty list situation more explicitly)...

Reply:
Anyway, I don't think there's a behavior change either.  But I'll triple-check from the front end once this is being triggered.

Reply:
The only reason I can think of is that a JSON-encoded value is ready to be written right to the database. But I think JSON-encoding makes more sense _right before_ writing to the database, not while the controller might still be validating or working with the value. A task for later!